### PR TITLE
Add application.rename method

### PR DESCRIPTION
**Pull request by nghiant2710**
Connects-to: https://github.com/balena-io/balena-sdk/pull/1002
Change-type: minor
Signed-off-by: Trong Nghia Nguyen <nghiant2710@gmail.com>

### CODE REVIEW

**Comment by nghiant2710** [comment]
@balena-ci test

File: balena/models/application.py [hunk 1]
```diff
@@ -414,6 +414,33 @@ def remove(self, name):
             'application', 'DELETE', params=params,
             endpoint=self.settings.get('pine_endpoint'), login=True
         )
+        
+    def rename(self, app_id, new_name):
+        """
+        Rename application. This function only works if you log in using credentials or Auth Token.
+
+        Args:
+            app_id (str): application id.
```
Comment (thgreasi) [inline balena/models/application.py]: Shouldn't this be "number" instead of `str`?

**Review by thgreasi — COMMENTED** [review]
LGTM, feel free to self-certify

**Comment by nghiant2710** [comment]
@balena-ci I self-certify!

**Pull request merged**